tvshowmatching: match xx.xx.xxxx as dd.mm.yyyy

Rechi opened the pull request:
<!--- Provide a general summary of your change in the Title above -->

## Description
tvshowmatching: match xx.xx.xxxx as dd.mm.yyyy
dot separated date format normally is interpreted as dd.mm.yyyy instead of mm.dd.yyyy
<!--- Describe your change in detail -->

## Motivation and Context
fix https://trac.kodi.tv/ticket/17592
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
Scraped dummy files with different date formats
- `The Daily Show/05.09.2017.strm`
- `The Daily Show/2017.09.05.strm`
- `The Daily Show/2017-09-05.strm`
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

<!--## Screenshots (if appropriate):-->

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

Razzeee commented:
So  basically we're moving from american formatting to european formatting?

Rechi commented:
I thought the separator would be `/` if date format is `mm/dd/yyyy` and not `.`.

Razzeee commented:
You're right about that. Nevermind.

On Sat, 23 Sep 2017, 21:10 Rechi <notifications@github.com> wrote:

> I thought the separator would be / if date format is mm/dd/yyyy and not ..
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/xbmc/xbmc/pull/12807#issuecomment-331663672>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFqyZOHhvHNeMlXbZtEER-FPQXtcdavQks5slVeIgaJpZM4Pa1HR>
> .
>

Razzeee commented:
Or well can you use / in file names?

On Sat, 23 Sep 2017, 21:15 Kolja <razzeee@gmail.com> wrote:

> You're right about that. Nevermind.
>
> On Sat, 23 Sep 2017, 21:10 Rechi <notifications@github.com> wrote:
>
>> I thought the separator would be / if date format is mm/dd/yyyy and not .
>> .
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/xbmc/xbmc/pull/12807#issuecomment-331663672>, or mute
>> the thread
>> <https://github.com/notifications/unsubscribe-auth/AFqyZOHhvHNeMlXbZtEER-FPQXtcdavQks5slVeIgaJpZM4Pa1HR>
>> .
>>
>

Rechi commented:
@Razee no, `/` can't be used in filenames.
Would be using `-` an alternative or should we leave it as it is?

Razzeee commented:
Don't really have any oppinion about that. Using minus would work I guess, but maybe we should ask our americans if there is any format that they would expect in this case.

Rechi commented:
@Razzeee whom should I ping?

MartijnKaijser commented:
@mkortstiege @notspiff you know something about scrapers correct?

phunkyfish commented:
@Rechi is this something you are still interested in merging?

phunkyfish commented:
@Rechi what do you need to progress this?

thezoggy commented:
daily or air-by-date format is usually seen in `yyyy.mm.dd` or `yyyy-mm-dd` and sometimes without any separator. 

sharing regex from another app on how they match these:
```
(?<airyear>19[6-9]\d|20\d\d)(?<sep>[-_]?)(?<airmonth>0\d|1[0-2])\k<sep>(?<airday>[0-2]\d|3[01])(?!\d)
```

Rechi committed:
tvshowmatching: restrict month and day values to 00-39

this is still a too large range, but its smaller than 00-99

Rechi committed:
tvshowmatching: match xx.xx.xxxx as dd.mm.yyyy

dot separated date format normally is interpreted as dd.mm.yyyy instead of mm.dd.yyyy
